app/src/apdu_handler.c

```diff
@@ -203,15 +265,20 @@ void handleApdu(volatile uint32_t *flags, volatile uint32_t *tx, uint32_t rx) {
         }
         CATCH(EXCEPTION_IO_RESET) { THROW(EXCEPTION_IO_RESET); }
         CATCH_OTHER(e) {
-            switch (e & 0xF000) {
-                case 0x6000:
-                case APDU_CODE_OK:
-                    sw = e;
-                    break;
-                default:
-                    sw = 0x6800 | (e & 0x7FF);
-                    break;
+            if (e == APDU_CODE_LEGACY_PARSER_ERROR) {
+                sw = e;
+            } else {
+                switch (e & 0xF000) {
+                    case 0x6000:
+                    case APDU_CODE_OK:
+                        sw = e;
+                        break;
+                    default:
+                        sw = 0x6800 | (e & 0x7FF);
+                        break;
+                }
```

Comment:
Are we using their error codes? It is not that clear to me what we are doing here. Maybe defining these values as const somewhere would make it easier to follow

Reply:
This is not related to their error. I added APDU_CODE_LEGACY_PARSER_ERROR to identify if we encounter issues when building the input buffer with invalid JSON in legacy code. This also allows us to test all edge cases in the logic of legacy_handleSignTransaction with different data sizes